### PR TITLE
Support lookup indexScan using IN expression as filter

### PR DESCRIPTION
**Pull request by Aiee**
1. Fix constant fold for container expressions
2. Convert `IN` expression to `OR` expression in `LOOKUP` if the property in the`IN` expression has a valid index:
    - Stand alone `IN` expr: `A in [B, C, D]  =>  (A == B) or (A == C) or (A == D)`
    - With `AND` expr : `A in [a, b] AND B  =>  (A==a AND B) OR (A==b AND B)`
    - With `OR` expr: `A in [a, b] OR B  =>   A==a OR A==b OR B`
3. If the container has only one element, the `IN` expression will be transformed to a `relEQ` expression. e.g.  `A in [B]  =>  A==B`

### CODE REVIEW

File: tests/tck/features/lookup/TagIndexFullScan.feature [hunk 1]
```diff
@@ -125,11 +125,10 @@ Feature: Lookup tag index full scan
...
       | "Boris Diaw"        | 36         |
     And the execution plan should be:
-      | id | name             | dependencies | operator info                            |
-      | 3  | Project          | 2            |                                          |
-      | 2  | Filter           | 4            | {"condition": "(player.age IN [39,36])"} |
-      | 4  | TagIndexFullScan | 0            |                                          |
-      | 0  | Start            |              |                                          |
+      | id | name      | dependencies | operator info |
+      | 3  | Project   | 4            |               |
+      | 4  | IndexScan | 0            |               |
```
Comment (czpmango) [inline tests/tck/features/lookup/TagIndexFullScan.feature:130]: Add columnHints check.

Comment (Aiee) [reply tests/tck/features/lookup/TagIndexFullScan.feature:130]: TCK doesn't support multiple colHints.

File: tests/tck/features/lookup/TagIndexFullScan.feature [hunk 2]
```diff
@@ -125,11 +125,69 @@ Feature: Lookup tag index full scan
...
+      LOOKUP ON player WHERE player.age IN [40, 25] AND player.name IN ["ABC", "Kobe Bryant"] YIELD player.age
+      """
+    Then the result should be, in any order:
+      | VertexID      | player.age |
+      | "Kobe Bryant" | 40         |
+    And the execution plan should be:
+      | id | name             | dependencies | operator info                                                                                                            |
+      | 3  | Project          | 2            |                                                                                                                          |
+      | 2  | Filter           | 4            | {"condition": "(((player.age==40) OR (player.age==25)) AND ((player.name==\"ABC\") OR (player.name==\"Kobe Bryant\")))"} |
+      | 4  | TagIndexFullScan | 0            |                                                                                                                          |
```
Comment (jievince) [inline tests/tck/features/lookup/TagIndexFullScan.feature:189]: What's the actual operator info of TagIndexFullScan? Could you paste it here?

Comment (Aiee) [reply tests/tck/features/lookup/TagIndexFullScan.feature:189]: `TagIndexFullScan` scans the whole tag, so the `"columnHints"` is empty.
```
|  4 | TagIndexFullScan | 0            | ver: 0, rows: 56, execTime: 0us, totalTime: 9707us | outputVar: [                                                                                                   |
|    |                  |              |                                                    |   {                                                                                                            |
|    |                  |              |                                                    |     "colNames": [                                                                                              |
|    |                  |              |                                                    |       "VertexID",                                                                                              |
|    |                  |              |                                                    |       "player.age",                                                                                            |
|    |                  |              |                                                    |       "player.name"                                                                                            |
|    |                  |              |                                                    |     ],                                                                                                         |
|    |                  |              |                                                    |     "type": "DATASET",                                                                                         |
|    |                  |              |                                                    |     "name": "__TagIndexFullScan_1"                                                                             |
|    |                  |              |                                                    |   }                                                                                                            |
|    |                  |              |                                                    | ]                                                                                                              |
|    |                  |              |                                                    | inputVar:                                                                                                      |
|    |                  |              |                                                    | space: 1                                                                                                       |
|    |                  |              |                                                    | dedup: false                                                                                                   |
|    |                  |              |                                                    | limit: 9223372036854775807                                                                                     |
|    |                  |              |                                                    | filter:                                                                                                        |
|    |                  |              |                                                    | orderBy: []                                                                                                    |
|    |                  |              |                                                    | schemaId: 2                                                                                                    |
|    |                  |              |                                                    | isEdge: false                                                                                                  |
|    |                  |              |                                                    | returnCols: [                                                                                                  |
|    |                  |              |                                                    |   "_vid",                                                                                                      |
|    |                  |              |                                                    |   "age",                                                                                                       |
|    |                  |              |                                                    |   "name"                                                                                                       |
|    |                  |              |                                                    | ]                                                                                                              |
|    |                  |              |                                                    | indexCtx: [                                                                                                    |
|    |                  |              |                                                    |   {                                                                                                            |
|    |                  |              |                                                    |     "columnHints": [],                                                                                         |
|    |                  |              |                                                    |     "filter": "",                                                                                              |
|    |                  |              |                                                    |     "index_id": 8                                                                                              |
|    |                  |              |                                                    |   }                                                                                                            |
|    |                  |              |                                                    | ]                                                                                                              |
```

Comment (jievince) [reply tests/tck/features/lookup/TagIndexFullScan.feature:189]: So why does the condition of `filter` not be pushed (or be partitially pushed) into `IndexScan`?

**Comment by Aiee** [comment]
This PR will be moved to vesoft-nebula repo.

**Comment by Aiee** [comment]
This PR has been transferred to https://github.com/vesoft-inc/nebula/pull/2906

**Pull request closed without merging**